src/lib.rs

```diff
@@ -257,20 +257,12 @@ impl Command {
     }
 }
 
-/// This non-public module is used to wrap public traits used inside the crate,
-/// which should not be public to the user.
-///
-/// This helps getting around the "can't leak private trait" error message.
-mod somewhat_private_traits {
-    use super::PowerMode;
-
-    pub trait MeasurementDuration {
-        /// Return the max measurement duration (depending on the mode) in
-        /// microseconds.
-        fn max_measurement_duration(mode: PowerMode) -> u16;
-    }
+/// Determine the max measurement duration (according to the datasheet).
+pub trait MeasurementDuration {
+    /// Return the max measurement duration (depending on the mode) in
+    /// microseconds.
+    fn max_measurement_duration(mode: PowerMode) -> u16;
```

Comment:
Could this be a `const fn`? I'm not sure if it is allowed in a trait definition.

Reply:
```
error[E0379]: functions in traits cannot be declared const
   --> src/lib.rs:264:5
    |
264 |     const fn max_measurement_duration(mode: PowerMode) -> u16;
    |     ^^^^^ functions in traits cannot be const

error[E0658]: const fn is unstable
   --> src/lib.rs:264:5
    |
264 |     const fn max_measurement_duration(mode: PowerMode) -> u16;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #57563 <https://github.com/rust-lang/rust/issues/57563> for more information
```

Reply:
Ah I suspected something like that, but wasn't sure. Thanks for verifying :slightly_smiling_face: